src/QsCompiler/CompilerOptimizations/Utils.fs

```diff
@@ -166,50 +74,34 @@ and private defaultValue (bt: TypeKind): Expr option =
     | ArrayType t -> constructNewArray t.Resolution 0
     | _ -> None
 
-/// Wraps a QsExpressionType in a basic TypedExpression.
-/// The returned TypedExpression has no type param / inferred info / range information,
-/// and it should not be used for any code step that requires this information.
-and internal wrapExpr (expr: Expr) (bt: TypeKind): TypedExpression =
-    let ii = {IsMutable=false; HasLocalQuantumDependency=false}
-    TypedExpression.New (expr, ImmutableDictionary.Empty, ResolvedType.New bt, ii, Null)
-
 
-/// Counts the number of MissingExprs in the given tuple
-let rec private countMissingExprs (expr: TypedExpression): int =
+/// Returns true if the given expression contains any MissingExprs
+let rec internal hasMissingExprs (expr: TypedExpression): bool =
```

Comment:
How about splitting out containsMissing and IsPartialApplication (SyntaxExtensions.fs) into it's own method and using that instead.

Reply:
Yeah, this would be a good change